src/main/java/teammates/ui/webapi/CreateAccountRequestAction.java

```diff
@@ -1,56 +1,209 @@
+
+    @Override
+    AuthType getMinAuthLevel() {
+        return AuthType.PUBLIC;
+    }
+
+    @Override
+    void checkSpecificAccessControl() throws UnauthorizedAccessException {
+        AccountRequestCreateIntent intent =
+                AccountRequestCreateIntent.valueOf(getNonNullRequestParamValue(Const.ParamsNames.INTENT));
+
+        switch (intent) {
+        case ADMIN_CREATE:
+            if (userInfo == null || !userInfo.isAdmin) {
+                throw new UnauthorizedAccessException("Admin privilege is required to access this resource.");
+            }
+            break;
+        case PUBLIC_CREATE:
+            break;
+        default:
+            throw new InvalidHttpParameterException("Unknown intent " + intent);
+        }
+    }
 
     @Override
     public JsonResult execute() throws InvalidHttpRequestBodyException, InvalidOperationException {
-        AccountCreateRequest createRequest = getAndValidateRequestBody(AccountCreateRequest.class);
+        AccountRequestCreateIntent intent =
+                AccountRequestCreateIntent.valueOf(getNonNullRequestParamValue(Const.ParamsNames.INTENT));
+
+        if (!intent.equals(AccountRequestCreateIntent.ADMIN_CREATE)) {
+            String userCaptchaResponse = getRequestParamValue(Const.ParamsNames.USER_CAPTCHA_RESPONSE);
+            if (userCaptchaResponse == null || !recaptchaVerifier.isVerificationSuccessful(userCaptchaResponse)) {
+                throw new InvalidHttpParameterException("Please check the \"I'm not a robot\" box before submission.");
+            }
+        }
+
+        String type = getRequestParamValue(Const.ParamsNames.ACCOUNT_REQUEST_TYPE);
+        if (type == null || AccountRequestType.valueOf(type) != AccountRequestType.INSTRUCTOR_ACCOUNT) {
+            throw new InvalidHttpParameterException("Only instructor accounts can be created.");
+        }
+
+        AccountRequestCreateRequest createRequest = getAndValidateRequestBody(AccountRequestCreateRequest.class);
 
         String instructorName = createRequest.getInstructorName().trim();
+        String instructorInstitute = createRequest.getInstructorInstitute().trim();
+        String instructorCountry = createRequest.getInstructorCountry().trim();
         String instructorEmail = createRequest.getInstructorEmail().trim();
-        String instructorInstitution = createRequest.getInstructorInstitution().trim();
+        String instructorHomePageUrl = createRequest.getInstructorHomePageUrl().trim();
+        String comments = createRequest.getComments().trim();
 
-        AccountRequestAttributes accountRequestToCreate = AccountRequestAttributes
-                .builder(instructorEmail, instructorInstitution, instructorName)
-                .build();
+        AccountRequestAttributes accountRequestToCreate;
         AccountRequestAttributes accountRequestAttributes;
 
+        AccountRequestCreateErrorResults errorResults = new AccountRequestCreateErrorResults();
+
         try {
```

Comment:
Try blocks should be narrowly scoped.

Reply:
That results in duplicate catch part. I don't think there's big problem with the current approach.

Reply:
Changed.